azure-pipelines.yml

```diff
@@ -311,3 +311,21 @@ jobs:
   - bash: ci/run_win.sh
     displayName: 'Testing'
 
+- job: PyPi
+  dependsOn:
+    - 'Ubuntu16Py37'
+    - 'Ubuntu16All'
+    - 'Ubuntu16Py27'
+    - 'Ubuntu16PyPy2'
+    - 'Ubuntu16PyPy3'
+    - 'OSX10_13'
+    - 'Win2016'
+  condition: and(succeeded(), eq(variables['build.sourceBranch'], 'refs/heads/master'))
```

Comment:
This comment has nothing to do with this PR, but I like your way of expressing the conditions better, so maybe we can change the other conditions in the file as:
```and(succeeded(), or(eq(variables['build.sourceBranch'], 'refs/heads/master'), eq(variables['Build.Reason'], 'PullRequest')))```

Reply:
AFAIU the ``succeeded``` part only makes sense if we have a dependsOn. I can change the use of sourceBranch instead of sourceBranchName.